incubator/nodejs/image/project/Dockerfile

```diff
@@ -8,14 +8,29 @@ RUN rm -rf /project/user-app/node_modules && mkdir -p /project/user-app/node_mod
 
 # Install user-app dependencies
 WORKDIR /project/user-app
-RUN npm install --production 
+RUN npm install --production
+
+# Run a build phase. Projects that need to execute build commands can customize the
+# 'build' script in their package.json. The build script should call 'npm install'.
+# If no build is required, just install production dependencies.
```

Comment:
> The build script should call 'npm install'.
> If no build is required, just install production dependencies.

Why is this needed is we are running `npm install --production` ourselves on line 11?

Reply:
The idea behind these changes was to leave the build for existing apps unchanged - projects that do not define a 'build' and 'prune' script would build as before.

If a build script is created, and the build process requires additional dependencies (defined in `devDependencies`), they would need to be installed at this stage.  The existing `npm install --production` would not install the `devDependencies` that may be necessary for the build phase (for example, a Typescript project would only require the tsc compiler at build time, which emits standard Javascript for runtime).